CMakeLists.txt

```diff
@@ -17,23 +17,22 @@ if(MSVC)
 
 elseif(NOT APPLE)
     # Linux flags
-    string(APPEND CMAKE_CXX_FLAGS " -Wall -Wextra -Wl,-rpath,'$ORIGIN'")
+    string(APPEND CMAKE_CXX_FLAGS " -Wall -Wextra ")
     string(APPEND CMAKE_C_FLAGS " -Wall -Wextra")
 endif()
 
+option(BUILD_SHARED_LIBS "Whether to build shared libraries" ON)
+option(BUILD_PIPER_PHONMIZE_EXE "Whether to build piper_phonemize_exe" ON)
+option(BUILD_PIPER_PHONMIZE_TESTS "Whether to build tests" ON)
+
 add_library(
-    piper_phonemize SHARED
+    piper_phonemize
     src/phonemize.cpp
     src/phoneme_ids.cpp
     src/tashkeel.cpp
     src/shared.cpp
 )
 
-set_target_properties(piper_phonemize PROPERTIES
```

Comment:
Why did you remove this?

Reply:
piper-phonemize is used as a subproject in sherpa-onnx and it is always compiled from source along with sherpa-onnx. 

We don't need to manage the version of piper-phonemize via soname. 

In addition, setting soname will produce two more symlink files. If I use `cp /path/to/install/lib/lib*.so* /some/dest/dir`, it will dereference the symlink files and copy the real file, which means there are two more copies of the lib.

I don't see the need to keep the soname. I cannot think of a use case where we only update the version of
piper-phonemize without updating the main project where piper-phonemize is used.

Reply:
Other people have requested this, so I'd like to keep it. We could make it an option, though.

Reply:
> Other people have requested this, so I'd like to keep it. We could make it an option, though.

Ok, I will create an option, defaulting to ON.